pkg/nuget/lock/parse.go

```diff
@@ -49,7 +51,33 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 				Version: packageContent.Resolved,
 			}
 			libs = append(libs, lib)
+
+			depId := utils.PackageID(packageName, packageContent.Resolved)
+			dependsOn := make([]string, 0)
+
+			for depName := range packageContent.Dependencies {
+				dependsOn = append(dependsOn, utils.PackageID(depName, targetContent[depName].Resolved))
+			}
+
+			if depsMap[depId] != nil {
+				dependsOn = append(dependsOn, depsMap[depId]...)
+				if dependsOn = utils.UniqueStrings(dependsOn); dependsOn == nil {
+					dependsOn = make([]string, 0)
+				}
+			}
+
+			depsMap[depId] = dependsOn
```

Comment:
We only need to keep dependencies with child dependencies([e.g. for npm](https://github.com/aquasecurity/go-dep-parser/blob/60502daef4babbc9eee9c57c524625f6f50891f4/pkg/nodejs/npm/parse_testcase.go#L16-L21)).

Reply:
Actually, that's what I do here :)

Reply:
you save all dependencies here.
e.g.:
```
...
             "pkgA": {
                "type": "Direct",
                "requested": "[1.0.0, )",
                "resolved": "1.0.0",
                "contentHash": ""
            },
            "pkgB": {
                "type": "Direct",
                "requested": "[2.0.0, )",
                "resolved": "2.0.0",
                "contentHash": "",
                "dependencies": {
                    "pkgC": "3.0.0",
                }
            },
           "pkgC": {
                "type": "Direct",
                "requested": "[3.0.0, )",
                "resolved": "3.0.0",
                "contentHash": ""
            },
...
```
For this lock file you must skip `pkgA` and `pkgC` dependencies, because they have no child deps.
DependOn result:
```
Dependency{
		{ID: "pkgB@2.0.0", DependsOn: []string{"pkgC@3.0.0"}},,
	}
```
I think something like this should be added:
```
if len(dependsOn) > 0 {
	depsMap[depId] = dependsOn
}
```